lxd/network/driver_macvlan.go

```diff
@@ -29,46 +29,53 @@ func (n *macvlan) Validate(config map[string]string) error {
 		// ---
 		//  type: string
 		//  shortdesc: Parent interface to create `macvlan` NICs on
+		//  scope: local
 		"parent": validate.Required(validate.IsNotEmpty, validate.IsInterfaceName),
 		// lxdmeta:generate(entities=network-macvlan; group=network-conf; key=mtu)
 		//
 		// ---
 		//  type: integer
 		//  shortdesc: MTU of the new interface
+		//  scope: global
```

Comment:
There maybe some confusion around the scope (pun intended) of this PR.

It is to update the metdata API with the current scope of each setting.

The definition for "local" scope in the context of the docs and metadata API is whether the config key in question can be defined on a per-cluster member basis.

As instance devices are always tied to an instance, the value of their settings applies irrespective to which cluster member they are running on, and thus instance and device settings are inherently global.

Reply:
I'm probably confused but I understood this `mtu` config to be for a `macvlan` bridge, not an instance's NIC device.

Reply:
Ah yeah true, i somehow saw this as for the macvlan device.

Indeed `parent` should likely be local indeed, as it is the network indirection concept that allows an instance NIC device referencing a network to take on per-cluster member settings.

So for this PR specifically we are really only interested on what is currently implemented as local/global, rather than potential future changes. 

@kadinsayani there are 2 variables you'll want to look at for this:

```go
// NodeSpecificNetworkConfig lists all network config keys which are node-specific.
var NodeSpecificNetworkConfig = []string{
	"bgp.ipv4.nexthop",
	"bgp.ipv6.nexthop",
	"bridge.external_interfaces",
	"parent",
}
```

and 

```go
// NodeSpecificStorageConfig lists all storage pool config keys which are node-specific.
var NodeSpecificStorageConfig = []string{
	"size",
	"source",
	"source.wipe",
	"volatile.initial_source",
	"zfs.pool_name",
	"lvm.thinpool_name",
	"lvm.vg_name",
}
```

Reply:
> So for this PR specifically we are really only interested on what is currently implemented as local/global, rather than potential future changes.

Yes indeed. But it's nice that Mark caught a discrepancy which will be followed through :)